src/cpp/session/modules/jobs/AsyncRJobManager.cpp

```diff
@@ -117,7 +121,7 @@ Error registerAsyncRJob(boost::shared_ptr<AsyncRJob> job,
    job->registerJob();
 
    // remove the job from the registry when it's done
-   job->setOnComplete([&]() 
+   job->addOnComplete([&]() 
```

Comment:
I wonder if this lambda should actually capture by value? IIUC it's capturing the job (a shared pointer) by reference, which means in theory if the job were to be deleted we'd be left with a dangling reference. Seems unlikely to happen in practice but capturing by value feels safer.

Reply:
Agreed! I've made this change.